test/unit/math/prim/fun/value_of_test.cpp

```diff
@@ -42,25 +42,19 @@ TEST(MathMatrixPrimArr, value_of) {
 
 TEST(MathFunctions, value_of_int_return_type_short_circuit) {
   std::vector<int> a(5, 0);
-  EXPECT_FALSE((std::is_same<decltype(stan::math::value_of(a)),
```

Comment:
Why do these tests need to be removed if the behaviour hasn't changed?

Reply:
They are kinda stupid. We already have tests that checks what return type of `value_of` is, we dont need to check what it is not. 

Also while behavour is more or less the same, exact return type is not.